### PR TITLE
feat(docs): add guide for hyperlinks and rich text in DeltaV

### PR DESCRIPTION
**Pull request by FelixNicolaeBucsa**
## Proposed Changes

_Adds callout for info on Rich Text and Hyperlinks in DeltaV._

## Types of changes

_What type of change does this pull request make (put an `x` in the boxes that apply)?_

- [x] Content update.
- [ ] Bug fix (non-breaking change that fixes an issue).
- [x] New feature added (non-breaking change that adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to stop working as expected).
- [x] Something else (e.g., tests, scripts, example, deployment, infrastructure).

## Checklist

_Put an `x` in the boxes that apply:_

- [x] I have read the [CONTRIBUTING](https://github.com/fetchai/docs/blob/main/CONTRIBUTING.md) guide
- [x] Checks and tests pass locally

## Further comments

_[if this is a relatively large or complex change, kick off a discussion by explaining why you chose the solution you did, what alternatives you considered, etc...]_

### CODE REVIEW

File: pages/concepts/ai-engine/deltav.mdx [hunk 1]
```diff
@@ -15,3 +15,17 @@ Start developing your Agents to encapsulate Agent Functions and register your ag
     <Callout type="info" emoji="ℹ️">
       Head to [DeltaV ↗️](https://deltav.agentverse.ai/), sign in and get started with your first request!
     </Callout>
+
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/concepts/ai-engine/deltav.mdx]: Unnecessary callout as already called out in previous page?

File: pages/guides/agents/intermediate/agent-functions.mdx [hunk 2]
```diff
@@ -139,6 +139,20 @@ A **Primary Function** is the primary Agent Function and for instance, it could
 
 Similarly, a **Secondary Function** is an Agent Sub-function providing secondary services that likely need additional context or information to carry out the Primary Function. _Secondary Functions are executed in combination with the Objective task_. If that's the case, DeltaV would see that the Agent Primary Function can be fulfilled by executing a Secondary Function, thus, it will contact this latter one which may or may not require gaining context directly from the user.
 
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/guides/agents/intermediate/agent-functions.mdx:144]: Update this to give a link to page on rich text

File: pages/guides/agents/intermediate/ai-engine-compatible-agent.mdx [hunk 3]
```diff
@@ -107,3 +107,17 @@ To enable communication from agents to the AI Engine, we import **3 objects** th
     <Callout type="warning" emoji="⚠️">
       Whenever operating on DeltaV, we encourage everyone to select the **Next Generation AI Engine** personality type. This AI Engine personality stands as a significant AI Engine personality type offering _enhanced scalability_, _reliability_, and _flexibility_. The major key features include advanced context understanding, improved function recommendations, and the ability to handle multiple dialogue formats.
     </Callout>
+
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/guides/agents/intermediate/ai-engine-compatible-agent.mdx]: let's remove this callout

File: pages/guides/agents/intermediate/primary-secondary-functions.mdx [hunk 4]
```diff
@@ -264,7 +264,21 @@ After your function has been executed you can see the `Agent Response` message.
 
 ![](../../../../src/images/guides/agentverse/Hugging-Face-System/deltaV-5.png)
 
-With that, **you have got a hugging face service which can be discovered and contacted with DeltaV. Awesome!**
+With that, **you have got a hugging face function which can be discovered and contacted with DeltaV. Awesome!**
+
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/guides/agents/intermediate/primary-secondary-functions.mdx]: let's remove this callout

File: pages/guides/agentverse/agentverse-functions/field-descriptions-for-deltav.mdx [hunk 5]
```diff
@@ -112,6 +112,20 @@ Ideal field descriptions for this example will be like:
 
 ![](src/images/guides/services/service-guide/business_finder_3.png)
 
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/guides/agentverse/agentverse-functions/field-descriptions-for-deltav.mdx]: let's remove this callout

File: pages/guides/agentverse/agentverse-functions/registering-agent-services.mdx [hunk 6]
```diff
@@ -171,3 +171,17 @@ With that, **you have got an Agent Function which can be discovered and contacte
     <Callout type="info" emoji="ℹ️">
       For an example on how to register a Local Agent Function on the Agentverse and subsequently retrieve it on DeltaV, check out this [guide ↗️](/guides/agents/running-locally) helping you to create a locally run Agent with an Agent Function, which is then registered on the Agentverse and retrievable on DeltaV!
     </Callout>
+
+    <Callout type="info" emoji="ℹ️">
```
Comment (devjsc) [inline pages/guides/agentverse/agentverse-functions/registering-agent-services.mdx]: let's remove this callout

**Review by devjsc — CHANGES_REQUESTED** [review]

File: pages/guides/agents/intermediate/ai-engine-compatible-agent.mdx (hunk 3 above)
Comment (FelixNicolaeBucsa) [reply pages/guides/agents/intermediate/ai-engine-compatible-agent.mdx]: Done

File: pages/guides/agents/intermediate/primary-secondary-functions.mdx (hunk 4 above)
Comment (FelixNicolaeBucsa) [reply pages/guides/agents/intermediate/primary-secondary-functions.mdx]: Done

File: pages/guides/agentverse/agentverse-functions/field-descriptions-for-deltav.mdx (hunk 5 above)
Comment (FelixNicolaeBucsa) [reply pages/guides/agentverse/agentverse-functions/field-descriptions-for-deltav.mdx]: Done

File: pages/guides/agentverse/agentverse-functions/registering-agent-services.mdx (hunk 6 above)
Comment (FelixNicolaeBucsa) [reply pages/guides/agentverse/agentverse-functions/registering-agent-services.mdx]: Done

File: pages/guides/agents/intermediate/agent-functions.mdx (hunk 2 above)
Comment (FelixNicolaeBucsa) [reply pages/guides/agents/intermediate/agent-functions.mdx:144]: Done

**Review by FelixNicolaeBucsa — COMMENTED** [review]
Have addressed comments. Let me know what you think.

**Comment by abhifetch** [comment]
Can you please update the code with below:

```
import requests
from ai_engine import UAgentResponse, UAgentResponseType

# Define a news content structure using the Model class to represent a news content
class NewsRequest(Model):
    company_name: str

# Create a protocol for the Hugging Face finbert agent
news_protocol = Protocol("Institute News")

async def fetch_symbol(company_name):
    url = f"https://www.alphavantage.co/query?function=SYMBOL_SEARCH&keywords={company_name}&apikey={ALPHA_VANTAGE_API_KEY}"
    response = requests.get(url)
    if response.status_code == 200:
        data = response.json()
        # Typically, the best match will be the first item in the bestMatches list
        if data.get('bestMatches') and len(data['bestMatches']) > 0:
            # Return the symbol of the best match
            symbol = data['bestMatches'][0]['1. symbol']
            return symbol
    return 'No Symbol found'

async def analyze_news(company_name):
    """Analyze news sentiment for the company using GNews."""
    url = f"https://gnews.io/api/v4/search?q={company_name}&token={GNEWS_API_KEY}&lang=en"
    response = requests.get(url)
    articles = response.json().get('articles', [])
    
    # Extract URLs and convert them to hyperlinks with titles
    button_list = []
    for article in articles:
        title = article.get("title", "No Title")
        url = article.get("url", "No Url")
        if url != "No Url":
            # Create a clickable button with the title as text
            button_list.append(f'<a href="{url}">{title}</a>')
    
    # Combine buttons into a single string with line breaks
    buttons_combined = '\n'.join(button_list)
    
    return buttons_combined

@news_protocol.on_message(model=NewsRequest, replies=UAgentResponse)
async def on_institute_news_request(ctx: Context, sender: str, msg: NewsRequest):
    ctx.logger.info(msg.company_name)
    symbol = await fetch_symbol(msg.company_name)
    ctx.logger.info(symbol)
    news_content = await analyze_news(symbol)
    ctx.logger.info(news_content)
    await ctx.send(sender, UAgentResponse(message=news_content, type=UAgentResponseType.FINAL))

agent.include(news_protocol)

```

also attach below output image for reference:

Screenshot 2024-08-20 at 10 21 04

**Pull request merged**